src/melee.cpp

```diff
@@ -1940,3 +1945,124 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
```

Comment:
This temporary is quite far from the place where it's used (furthermore, I'd inline it).